Python/executor.c

```diff
@@ -31,6 +31,39 @@
 #define ENABLE_SPECIALIZATION 0
 
 
+// TODO: continue editing this from tupleobject.c to fit _PyUOpExecutorObject
+static _PyUOpExecutorObject *
+tuple_alloc(Py_ssize_t size)
+{
+    _PyUOpExecutorObject *op = maybe_freelist_pop(size);
+    if (op == NULL) {
+        /* Check for overflow */
+        if ((size_t)size > ((size_t)PY_SSIZE_T_MAX - (sizeof(PyTupleObject) -
+                    sizeof(PyObject *))) / sizeof(PyObject *)) {
+            return (_PyUOpExecutorObject *)PyErr_NoMemory();
+        }
+        op = PyObject_GC_NewVar(_PyUOpExecutorObject, &PyTuple_Type, size);
+        if (op == NULL)
+            return NULL;
```

Comment:
These are the lines you want to keep. However, according to PEP 7, the if branch should have curly braces.

Reply:
Thank you very much for the help! 

I was looking into replacing PyTuple_Type and wasn't sure if it'd be necessary to recreate a PyTypeObject specific to this(and further potentially creating a new object) or if there's something better to use?